### PR TITLE
Store Sidebar: Refactor to enable state-connected menu items

### LINKED ISSUE #15869: WooCommerce orders: Add new order notification to sidebar navigation
When there are new orders to process the number should be displayed in the sidebar;

orders-number

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #15869

This PR refactors the store sidebar. Instead of generating it from the pages list in `getStorePages`, we "hard code" each item in the sidebar itself. This way we can add a state-connected Orders item, which displays a count of new orders (if any exist).

screen shot 2017-07-10 at 6 13 56 pm
screen shot 2017-07-10 at 6 13 48 pm

I noticed we're using `preloadSectionName=slug` in the generated `SidebarItem`s, do we have preloading set up? Should we be using an extension-scoped section name here? For example, `preloadSectionName=settings` loads the chunk for site settings. I've copied it over for this PR, but we should probably remove it before merging this (I don't think we need to worry about preloading yet...) — just looking for a second opinion.

**To test:**

- Load up this branch
- Click through pages and make sure links work, menu items are highlighted when they should be
- Make sure it works during setup, when only "Dashboard" should display

### CODE REVIEW

File: client/extensions/woocommerce/store-sidebar/index.js [hunk 1]
```diff
@@ -74,90 +64,120 @@ class StoreSidebar extends Component {
...
+					{ translate( 'Add' ) }
+				</SidebarButton>
+			</SidebarItem>
+		);
+	}
+
+	orders = () => {
+		const { orders, site, siteSuffix, translate } = this.props;
+		const link = '/store/orders' + siteSuffix;
+		// We don't use the addLink yet, but this ensures the item is selected on single views
```
Comment (justinshreve) [inline client/extensions/woocommerce/store-sidebar/index.js]: Should we call it `childLink/childLinks` like below so we don't need the comment?

File: client/extensions/woocommerce/store-sidebar/index.js [hunk 2]
```diff
@@ -74,90 +64,120 @@ class StoreSidebar extends Component {
...
+		} );
+
+		return (
+			<SidebarItem
+				className={ classes }
+				icon="house"
+				label={ translate( 'Dashboard' ) }
+				link={ link }
+				onNavigate={ this.onNavigate }
+				preloadSectionName={ 'dashboard' } />
```
Comment (justinshreve) [inline client/extensions/woocommerce/store-sidebar/index.js]: > do we have preloading set up

Nope.  I agree with your comment about removing these prior to merge.

File: client/extensions/woocommerce/store-sidebar/index.js [hunk 3]
```diff
@@ -74,90 +64,120 @@ class StoreSidebar extends Component {
...
+			orders: true,
+			'is-placeholder': ! site,
+			selected,
+		} );
+
+		return (
+			<SidebarItem
+				className={ classes }
+				icon="pages"
+				label={ translate( 'Orders' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/store-sidebar/index.js]: Hi! I've found a possible matching string that has already been translated **99** times:
`translate( 'Order' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/index.js [hunk 4]
```diff
@@ -36,123 +36,70 @@ const getStorePages = () => {
...
-				showDuringSetup: false,
-			},
+			path: '/store/products/:site',
 		},
 		{
 			container: ProductCreate,
 			configKey: 'woocommerce/extension-products',
-			path: '/store/product/:site',
-			parentPath: '/store/products/:site',
 			documentTitle: translate( 'New Product' ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/index.js]: Hi! I've found a possible matching string that has already been translated **61** times:
`translate( 'Product' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**